draft-ietf-oauth-transaction-tokens-03.md

```diff
@@ -278,6 +284,81 @@ In the diagram above, steps 1-5 are the same as in {{basic-flow}}
+                 o                   │              │
+             5   o    9              │              │
+               │ o ▲                 │              │
+               │   │                 │              ├──────────────┐
+               │   │                 │              │              │
+          ┌────▼───┴─────┐    6      │              │              │
+          │              ├───────────▶              ◀────────┐     │
+          │              │           │              │        │     │
+          │              ◀───────────               │        │     │
+          │              │     7     │              │        │     │
+          │   Internal   │           └──────────────┘     11 │     │12
+          │ Microservice │           ┌──────────────┐        │     │ 
+          │              │     8     │              │        │     │
+          │              ├───────────▶              │        │     │
+          │              │           │              ├────────┘     │
+          │              │           │              │              │
+          │              │           │   Workflow   │              │
+          │              │           │              ◀──────────────┘
+          │              │           │              │
+          └──────────────┘           │              │
+                                     │              │
+                                     └────┬───▲─────┘
+                                          │   │
+                                          ▼   │
+                                            o
+                                       13   o   14
+                                            o
+                                          │   ▲                   
+                                          │   │                        
+                                     ┌────▼───┴─────┐
+                                     │              │
+                                     │   Internal   │
+                                     │ Microservice │
+                                     │              │
+                                     └──────────────┘
+~~~
+{: #fig-arch-async-flow title="Batch Txn-Token Flow"}
+
+In the diagram above, steps 1-5 are the same as in {{basic-flow}}
+
+{:start="6"}
+
+6. An intermediate service determines that it needs to start a workflow. Once workflow is started successfully, then it can respond back to incoming request. The workflow will then can run for longer duration. The intermediate service needs to obtain a Batch Txn-Token to pass to the workflow as the workflow execution will outlive Txn-Token lifetime. It requests a Batch Txn-Token from the Txn-Token Service. It passes the incoming Txn-Token in the request, along with any additional context it needs to send the Txn-Token Service.
+7. The Txn-Token Service responds with a Batch Txn-Token
+8. The service that requested the Batch Txn-Token uses that Batch Txn-Token for invoking workflow
```

Comment:
Would it not be the responsibility of the workflow system to acquire/exchange the batch token? If the service creating the batch token is not the same as the one using it to create further tokens, there may be opportunities for breaches. At the very least the workflow/receiver would not be able to verify the batch token validity.